v03_pipeline/lib/tasks/write_new_variants_table.py

```diff
@@ -0,0 +1,257 @@
+            hl.bind(
+                lambda updates: hl.all(
+                    [
+                        updates.contains(
+                            hl.Struct(
+                                callset=callset_path,
+                                project_guid=project_guid,
+                            ),
+                        )
+                        for (
+                            callset_path,
+                            project_guid,
+                            _,
+                            _,
+                        ) in callset_project_pairs(
+                            self.callset_paths,
+                            self.project_guids,
+                            self.project_remap_paths,
+                            self.project_pedigree_paths,
+                        )
+                    ],
+                ),
+                hl.read_table(self.output().path).updates,
+            ),
+        )
+
+    def create_table(self) -> hl.Table:
+        callset_ht = get_callset_ht(
+            self.reference_genome,
+            self.dataset_type,
+            self.callset_paths,
+            self.project_guids,
+            self.project_remap_paths,
+            self.project_pedigree_paths,
+        )
+
+        # 1) Identify new variants.
+        annotations_ht = hl.read_table(
+            variant_annotations_table_path(
+                self.reference_genome,
+                self.dataset_type,
+            ),
+        )
+        new_variants_ht = callset_ht.anti_join(annotations_ht)
+
+        # 2) Join new variants against the reference dataset collections that are not "annotated".
+        for rdc in ReferenceDatasetCollection.for_reference_genome_dataset_type(
+            self.reference_genome,
+            self.dataset_type,
+        ):
+            if rdc.requires_annotation:
+                continue
+            rdc_ht = self.annotation_dependencies[f'{rdc.value}_ht']
+            new_variants_ht = new_variants_ht.join(rdc_ht, 'left')
+
+        # 3) TODO: Send to clingen allele registry.
+
+        # 4) Annotate new variants with VEP.
+        # Note about the repartition: our work here is cpu/memory bound and
+        # proportional to the number of new variants.  Our default partitioning
+        # will under-partition in that regard, so we split up our work
+        # with a partitioning scheme local to this task.
+        new_variants_count = new_variants_ht.count()
+        new_variants_ht = new_variants_ht.repartition(
+            constrain(
+                math.ceil(new_variants_count / VARIANTS_PER_VEP_PARTITION),
+                10,
+                10000,
+            ),
+        )
+        new_variants_ht = run_vep(
+            new_variants_ht,
+            self.dataset_type,
+            self.reference_genome,
+        )
+
+        # 5) Select down to the formatting annotations fields and any reference dataset collection annotations.
+        rdc_annotations = [
```

Comment:
This block got added because we've flipped the order of the join and the formatting annotations right?  Was there a reason for that?  

I don't think it matters too much... but probably slightly better for VEP to run before the join.  The join tends to be safe and VEP fragile... so running the fragile thing first means earlier failure.

Reply:
> This block got added because we've flipped the order of the join and the formatting annotations right? 

That's correct, and the order was flipped because I was going to rely on the rdc annotations to register alleles before vep, but now that I'm pretty sure registering will rely on vep, I will return the order to what it was before I created the new task.

Reply:
👍 yeah, I think that makes sense!